[Feature] Better error handling on invalid API keys

### Feature Description
We recently ran into a situation where a profile's user account was deleted in New Relic, which resulted in the CLI profile having an invalid API key. During automation runs in a CI/CD pipeline, attempts to execute `newrelic apm application search --name "<name>"` resulted in this error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x92b2e5]
goroutine 1 [running]:
github.com/newrelic/newrelic-cli/internal/apm.glob..func1(0x14dce80?, {0xdec319?, 0x2?, 0x2?})
	/home/runner/work/newrelic-cli/newrelic-cli/internal/apm/command_application.go:82 +0x3e5
github.com/spf13/cobra.(*Command).execute(0x14dce80, {0xc0002233c0, 0x2, 0x2})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0x14ea900)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:1040 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:968
main.Execute()
	/home/runner/work/newrelic-cli/newrelic-cli/cmd/newrelic/command.go:74 +0xed
main.main()
	/home/runner/work/newrelic-cli/newrelic-cli/cmd/newrelic/main.go:65 +0x1d
```

It took a day or so to identify that the NR user had been deleted and this error message was not very helpful.

### Describe Alternatives
It would be ideal if there was a clear and concise error message indicating that the failure was based on an authentication error; bonus points if it could pinpoint the failure to the invalid API key itself and the profile name being used by the CLI.

--- a/internal/apm/command_application.go
+++ b/internal/apm/command_application.go
@@ -79,8 +79,8 @@ The search command performs a query for an APM application name and/or account I
 				[]entities.EntitySearchSortCriteria{},
 			)
 
-			entityResults = results.Results.Entities
 			utils.LogIfFatal(err)
+			entityResults = results.Results.Entities
 		}
 
 		utils.LogIfFatal(output.Print(entityResults))

